engine/src/main/java/org/hibernate/validator/resourceloading/PlatformResourceBundleLocator.java

```diff
@@ -95,6 +162,21 @@ public PlatformResourceBundleLocator(String bundleName, ClassLoader classLoader,
 	 */
 	@Override
 	public ResourceBundle getResourceBundle(Locale locale) {
+		// we are in the preloading case
+		if ( !preloadedResourceBundles.isEmpty() ) {
+			// we need to use containsKey() as the cached resource bundle can be null
+			if ( preloadedResourceBundles.containsKey( locale ) ) {
```

Comment:
Maybe `preloadedResourceBundles.get(locale)` and then null check instead to remove additional map access ?

Reply:
You missed my comment: the bundle might be null and it is a valid value. Thus I really need the `containsKey()`.